cabal-install/Distribution/Client/CmdConfigure.hs

```diff
@@ -83,9 +85,18 @@ configureAction (configFlags, configExFlags, installFlags, haddockFlags)
     baseCtx <- establishProjectBaseContext verbosity cliConfig
 
     -- Write out the @cabal.project.local@ so it gets picked up by the
-    -- planning phase.
-    writeProjectLocalExtraConfig (distDirLayout baseCtx)
-                                 cliConfig
+    -- planning phase. If old config exists, then print the contents
+    -- before overwriting
+    exists <- doesFileExist "cabal.project.local"
+    if exists
+      then do
+        contents <- readFile "cabal.project.local"
+        putStr contents
```

Comment:
Although most tools encourage a trailing newline in file, it's by no means guaranteed. If the trailing newline is missing, you'll get weird output. Also, you might want the extra newline anyway to make the output look better. Hard to say without seeing the output.